sdk/keyvault/keyvault-admin/src/core/keyVaultClient.ts

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ */
+
+import * as coreHttp from "@azure/core-http";
+import * as Models from "./models";
+import * as Mappers from "./models/mappers";
+import * as Parameters from "./models/parameters";
+import * as operations from "./operations";
+import { KeyVaultClientContext } from "./keyVaultClientContext";
+
+class KeyVaultClient extends KeyVaultClientContext {
+  // Operation groups
+  roleDefinitions: operations.RoleDefinitions;
+  roleAssignments: operations.RoleAssignments;
+
+  /**
+   * Initializes a new instance of the KeyVaultClient class.
+   * @param apiVersion Client API version.
+   * @param credentials Subscription credentials which uniquely identify client subscription.
+   * @param [options] The parameter options
+   */
+  constructor(credentials: coreHttp.TokenCredential | coreHttp.ServiceClientCredentials, apiVersion: string, options?: coreHttp.ServiceClientOptions) {
+    super(credentials, apiVersion, options);
+    this.roleDefinitions = new operations.RoleDefinitions(this);
+    this.roleAssignments = new operations.RoleAssignments(this);
+  }
+
+  /**
+   * Creates a full backup using a user-provided SAS token to an Azure blob storage container.
+   * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
+   * @param [options] The optional parameters
+   * @returns Promise<Models.FullBackupResponse>
+   */
+  fullBackup(vaultBaseUrl: string, options?: Models.KeyVaultClientFullBackupOptionalParams): Promise<Models.FullBackupResponse>;
+  /**
+   * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
+   * @param callback The callback
+   */
+  fullBackup(vaultBaseUrl: string, callback: coreHttp.ServiceCallback<Models.FullBackupOperation>): void;
```

Comment:
I thought we were going to use v6 of the code generator to generate this package. Based on the callback pattern here it looks like we used v5. Any reason we could not use v6?

Reply:
I think I was confused, I will generate it with v6

Reply:
The v6 described in this issue is giving me an error: https://github.com/Azure/azure-sdk-for-js/issues/9590

```
Failed to install or start extension '@microsoft.azure/autorest.typescript' (6.0.0-dev.20200618.1)
  Error: Unable to resolve package '@microsoft.azure/autorest.typescript@6.0.0-dev.20200618.1'.
The script failed with exit code 1
```

I'll investigate tomorrow.

Reply:
@joheredi should be able to help here

Reply:
@sadasant v6's package was renamed to `@autorest/typescript` from the former `@microsoft.azure/autorest.typescript`, from the error message you shared above, it seems like you may have

```yaml
use-extension:
  "@microsoft.azure/autorest.typescript": "6.0.0-dev.20200618.1"
```

Could you please verify if your readme.md is using `@autorest/typescript`

```yaml
use-extension:
  "@autorest/typescript": "6.0.0-dev.20200623.2"
```

Reply:
Thank you Jose!!!

I did the changes and it didn't work 🤔.

Here's my diff:

```diff
diff --git a/sdk/keyvault/keyvault-admin/swagger/README.md b/sdk/keyvault/keyvault-admin/swagger/README.md
index 3e1bf75aa..4751fd2d3 100644
--- a/sdk/keyvault/keyvault-admin/swagger/README.md
+++ b/sdk/keyvault/keyvault-admin/swagger/README.md
@@ -6,7 +6,7 @@
 typescript:
   package-name: "@azure/keyvault-admin"
 use-extension:
-  "@microsoft.azure/autorest.typescript": "~5.0.1"
+  "@autorest/typescript": "6.0.0-dev.20200623.2"
 azure-arm: false
 generate-metadata: false
 add-credentials: false
```

Here's my error:

```
Failed to install or start extension '@autorest/typescript' (6.0.0-dev.20200623.2)
  Error: Unable to resolve package '@autorest/typescript@6.0.0-dev.20200623.2'.
The script failed with exit code 1
```

I'll write you over teams, if that helps!

Reply:
In a call with @joheredi we were able to make it work by adding ` --version=3.0.6267` to the package.json `regenerate` command! So I'll be updating this PR.